### PR TITLE
qsreplace: init at 0.0.3

### PR DESCRIPTION
**Pull request by averagebit**
## Description of changes

qsreplace - Accept URLs on stdin, replace all query string values with a user-supplied value

See:
    https://github.com/tomnomnom/qsreplace/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/qs/qsreplace/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildGoPackage,
+}:
```
Comment (dotlambda) [inline pkgs/by-name/qs/qsreplace/package.nix:5]:
```suggestion
{ lib
, fetchFromGitHub
, buildGoPackage
}:

```

File: pkgs/by-name/qs/qsreplace/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/tomnomnom/qsreplace";
+    description = "Accept URLs on stdin, replace all query string values with a user-supplied value, only output each combination of query string parameters once per host and path.";
+    mainProgram = "qsreplace";
+    maintainers = with maintainers; [averagebit];
+    platforms = platforms.unix;
+    sourceProvenance = with sourceTypes; [fromSource binaryNativeCode];
+    license = with licenses; [mit];
```
Comment (dotlambda) [inline pkgs/by-name/qs/qsreplace/package.nix:26]:
```suggestion
    license = licenses.mit;
```

File: pkgs/by-name/qs/qsreplace/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    repo = "qsreplace";
+    rev = "v${version}";
+    hash = "sha256-j9bqO2gp4RUxZHGBCIxI5nA3nD1dG4nCpJ1i4TM/fbo=";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/tomnomnom/qsreplace";
+    description = "Accept URLs on stdin, replace all query string values with a user-supplied value, only output each combination of query string parameters once per host and path.";
+    mainProgram = "qsreplace";
+    maintainers = with maintainers; [averagebit];
```
Comment (dotlambda) [inline pkgs/by-name/qs/qsreplace/package.nix:23]:
```suggestion
    maintainers = with maintainers; [ averagebit ];
```

File: pkgs/by-name/qs/qsreplace/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    platforms = platforms.unix;
+    sourceProvenance = with sourceTypes; [fromSource binaryNativeCode];
```
Comment (dotlambda) [inline pkgs/by-name/qs/qsreplace/package.nix:25]: Why `binaryNativeCode`?

File: pkgs/by-name/qs/qsreplace/package.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "tomnomnom";
+    repo = "qsreplace";
+    rev = "v${version}";
+    hash = "sha256-j9bqO2gp4RUxZHGBCIxI5nA3nD1dG4nCpJ1i4TM/fbo=";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/tomnomnom/qsreplace";
+    description = "Accept URLs on stdin, replace all query string values with a user-supplied value, only output each combination of query string parameters once per host and path.";
```
Comment (dotlambda) [inline pkgs/by-name/qs/qsreplace/package.nix:21]:
```suggestion
    description = "Accept URLs on stdin, replace all query string values with a user-supplied value";
```

**Review by dotlambda — COMMENTED** [review]
Usually there's no `v` in the commit message.

File: pkgs/by-name/qs/qsreplace/package.nix (hunk 4 above)
Comment (averagebit) [reply pkgs/by-name/qs/qsreplace/package.nix:25]: Oversight, on my part, should not be there.

Comment (averagebit) [inline pkgs/by-name/qs/qsreplace/package.nix:25]:
```suggestion
    sourceProvenance = sourceTypes.fromSource;
```

Comment (dotlambda) [reply pkgs/by-name/qs/qsreplace/package.nix]: It's supposed to be a list: https://github.com/NixOS/rfcs/blob/master/rfcs/0089-collect-non-source-package-meta.md#detailed-design

**Comment by dotlambda** [comment]
Please squash.

**Comment by dotlambda** [comment]
See https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-create-pull-requests for how to rebase and squash.

**Comment by raspher** [comment]
Result of `nixpkgs-review pr 284660` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qsreplace</li>
  </ul>
</details>

**Pull request merged**